File: backend/src/main/java/com/example/backend/models/Advertisement.java
package com.example.backend.models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Advertisement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private byte[] photo;
    private String name;
    private String description;
    private Boolean status_active;
    private Category category;
   // private User owner;

}

File: app/src/main/java/com/example/dropletbarterapp/models/Advertisement.kt
package com.example.dropletbarterapp.models

import com.google.gson.annotations.SerializedName

class Advertisement(
    @SerializedName("id") val id: Long,
    @SerializedName("photo") val photo: String,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String? = null,
    @SerializedName("statusActive") val statusActive: Boolean,
    @SerializedName("category") val category: Int,
    @SerializedName("ownerId") val ownerId: Long
) {
    //val tags = TagGenerator.generateTags(this)
}